sdk/src/main/java/com/supenta/flitchio/sdk/FlitchioController.java

```diff
@@ -121,15 +123,19 @@ public void onServiceConnected(ComponentName className, IBinder service) {
                         FlitchioLog.e("Unexpected error: could not authenticate to service.");
                     }
 
+                    if (listenerHandler != null) {
+                        listenerHandler.post(new StatusRunnable(flitchioService.isConnected(authToken)));
+                    }
+
                 } catch (RemoteException e) {
                     FlitchioLog.e("Unexpected error: could not identify this controller.");
                 }
             }
 
             // We connect the client in case he asked for it while binding was not ready. The client
-            // should not call onResume() if its listener is null, but in case he did, we don't
+            // should not call onResume() if its eventListener is null, but in case he did, we don't
             // register the client for a null Listener.
-            if (activityLifecycleMoment == ActivityLifecycle.ON_RESUME && listener != null) {
+            if (activityLifecycleMoment == ActivityLifecycle.ON_RESUME && eventListener != null) {
```

Comment:
Here the condition should include the statusListener as well. I think:
`if (activityLifecycleMoment == ActivityLifecycle.ON_RESUME && (eventListener != null || statusListener != null))`